### PR TITLE
add -c --cancel flag to siac allowance

### PR DESCRIPTION
**Pull request by johnathanhowell**

### CODE REVIEW

**Comment by lukechampine** [comment]
I feel like this makes more sense as a subcommand. It's a bit weird because we use `siac renter setallowance` to change the allowance, but `siac renter allowance cancel` seems most natural for canceling. Hmm.

**Comment by DavidVorick** [comment]
I wasn't a huge fan of `setallowance` in the first place.

**Comment by johnathanhowell** [comment]
Latest commit changes this to a subcommand.

File: siac/rentercmd.go [hunk 1]
```diff
@@ -257,6 +255,15 @@ func renterallowancecmd() {
 `, currencyUnits(allowance.Funds), allowance.Period)
 }
 
+// renterallowancecancelcmd cancels the current allowance.
+func renterallowancecancelcmd() {
+	err := post("/renter", "hosts=0&funds=0&period=0&renewwindow=0")
+	if err != nil {
+		die("error cancelling allowance:", err)
+	}
+	fmt.Println("successfully cancelled allowance.")
```
Comment (lukechampine) [inline siac/rentercmd.go]: Should be capitalized. "Allowance cancelled" is fine.

**Comment by lukechampine** [comment]
LGTM

**Pull request merged**